File: P349 Intersection of Two Arrays/Solution.java
public class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        int length1 = nums1.length;
        int length2 = nums2.length;
        Set<Integer> set = new HashSet<>();
        Set<Integer> intersect = new HashSet<>();
        for(int i = 0; i < length1; i++) {
            set.add(nums1[i]);
        }
        for(int i = 0; i < length2; i++) {
            if (set.contains(nums2[i])) {
                intersect.add(nums2[i]);
            }
        }
        int[] result = new int[intersect.size()];
        int i = 0;
        for (Integer num : intersect) {
            result[i++] = num;
        }
        return result;
    }
}

File: P349 Intersection of Two Arrays/Solution.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        std::set<int> set;
        std::set<int> intersect;

        for(int i=0;i<nums1.size();i++) {
            set.insert(nums1[i]);
        }
        for(int i=0;i<nums2.size();i++) {
            if (set.find(nums2[i]) != set.end()) {
                intersect.insert(nums2[i]);
            }
        }
        
        vector<int> result;
        std::set<int>::iterator it;
        for (it = intersect.begin(); it != intersect.end(); it++) {
            result.push_back(*it);
        }
        return result;
    }
};